CMakeLists.txt

```diff
@@ -102,10 +111,68 @@ find_package(Boost COMPONENTS filesystem)
 include_directories(${Boost_INCLUDE_DIRS})
 
 # TODO: utils-urdf should only be required for aikido::perception.
-find_package(DART REQUIRED COMPONENTS optimizer-nlopt utils-urdf CONFIG)
+find_package(DART 6.1.0 REQUIRED COMPONENTS optimizer-nlopt utils-urdf CONFIG)
 include_directories(${DART_INCLUDE_DIRS})
 
 #==============================================================================
+# Compiler settings
+#
+if(CMAKE_COMPILER_IS_GNUCXX)
+
+  if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS 4.8)
+    message(FATAL_ERROR "The installed g++ version is
+        ${CMAKE_CXX_COMPILER_VERSION}. ${PROJECT_NAME} requires g++ 4.8 or
+        greater."
+    )
+  endif()
+
+  if(TREAT_WARNINGS_AS_ERRORS)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror")
+  endif()
+
+elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+
+  if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS 3.3)
+    message(FATAL_ERROR "The installed Clang version is
+        ${CMAKE_CXX_COMPILER_VERSION}. ${PROJECT_NAME} requires clang 3.3 or
+        greater."
+    )
+  endif()
```

Comment:
All the compiler requirements are to ensure that C++ is supported by them. I'll add the comment.